core/challenges.go

```diff
@@ -24,20 +18,10 @@ func SimpleHTTPChallenge() Challenge {
 
 // DvsniChallenge constructs a random DVSNI challenge
 func DvsniChallenge() Challenge {
-	nonce := make([]byte, 16)
-	_, err := rand.Read(nonce)
-
-	if err != nil {
-		audit := blog.GetAuditLogger()
-		// AUDIT[ Error Conditions ] 9cc4d537-8534-4970-8665-4b382abe82f3
-		audit.EmergencyExit(err.Error())
-	}
-
 	return Challenge{
 		Type:   ChallengeTypeDVSNI,
 		Status: StatusPending,
-		R:      RandomString(32),
-		Nonce:  hex.EncodeToString(nonce),
+		Token:  NewToken(),
```

Comment:
These changes would be easier to review if you listed which spec changes were involved. Neither the pull request nor the original issue mention them. We have commits for them, so maybe they would be instructive (even a range of commits).